frontend/src/components/signup.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import ReactDOM from 'react-dom/client';
+import { useState } from "react"; 
+import axios from 'axios';
+import { fchmod } from 'fs';
+import '../components/signup.css'
+
+export const SignUp = () => {
+
+    const [firstName, setFirstName] = useState<string>('');
+    const [lastName, setLastName] = useState<string>('');
+    const [email, setEmail] = useState<string>('');
+    const [password, setPassword] = useState<string>('');
+    const [confirmPassword, setConfirmPassword] = useState<string>('');
+    const [warning, setWarning] = useState<string>(''); 
+
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+
+        e.preventDefault(); 
+
+            if(password === confirmPassword) {
+                setWarning('');
+                await axios.post('http://localhost:8000/signup', {
+                    firstName,
+                    lastName,
+                    email,
+                    password, 
+                })
+        
+                .then((res) => {
+                    console.log(res.data)
+                })
+        
+                .catch((err) => {
+                    setWarning('Please use a different email.');
+                })
+            }
+            
+            else{
+                setConfirmPassword(''); 
+                setWarning('Your password did not match. Please try again.')
+            }
+    }
+
+    return (
+
+        <section className = "container"> 
+        {/* <img className = "weWrapLogo" src = './weWrapGreen.png' alt = 'wewrap logo'> </img> */}
+        <header> 
+        <h3 className = "weWrapTitle"> Sign Up for WeWrap </h3>
+        <p className = "createAccount" > Create a free account or &nbsp;<a href = " ">login </a> </p>
+        </header>
+            {/* <img src = ""> WeWrap Logo </img> */}
+            <form className = "signUpForm" onSubmit = {handleSubmit}> 
+                <label htmlFor = "errorNotification"> {warning} </label>
+                <label htmlFor = "firstName"> 
+                    <p className = "firstNameTitle"> First Name * </p>
+                    <input className = "firstName" type = "text" required value = {firstName} onChange = {(event) => setFirstName(event.target.value)}/>
+                </label>
+                <label htmlFor = "lastName">
+                <p className = "lastNameTitle"> Last Name * </p>
+                    <input className = "lastName" type = "text" required value = {lastName} onChange={(event) => setLastName(event.target.value)}/> 
+                </label>
+                <label htmlFor = "email"> 
+                    <p className = "emailTitle"> Email * </p>
+                    <input className = "email" type = "text" required value = {email} onChange={(event) => setEmail(event.target.value)}/>
```

Comment:
Do we have plans to do email validation? i.e. making sure that an email is actually being provided rather than just a username like chinkevin25

Reply:
something else to think about is -- what happens if i tried to maliciously "take" someone elses email. Is that something we should be controlling for? 

What happens if i sign up with jesshe@gmail.com (no clue what your email is)

Reply:
could validation & malicious signing up be resolved with a sent email verification ?

Reply:
Yeah, it's actually kind of an involved process and many ways of doing it. We can leave it as it is for now since I want us to move past Auth soon! But the ideal way of actually doing this would be something like

"Enter Email" hit Submit 
weWrap will them E-Mail the email provided with a link to the rest of signup
User Clicks on a link in the email to continue the rest of the signup process (setting a password)

This way we can be sure that the email is actually the user rather than any one.

Alternatively, we could do whats considered magic link login-- but we dont need to do that for this project. This is where we don't even collect passwords -- we just email them a link and once they click on it, we consider that as signed in.